deps/obs-scripting/obs-scripting-python.c

```diff
@@ -1686,15 +1686,15 @@ bool obs_scripting_load_python(const char *python_path)
 
 #if RUNTIME_LINK
 	if (python_version.major == 3 && python_version.minor < 7) {
-#elif PY_VERSION_HEX < 0x030700b0
-	if (true) {
-#else
-	if (false) {
-#endif
 		PyEval_InitThreads();
 		if (!PyEval_ThreadsInitialized())
 			return false;
 	}
+#elif PY_VERSION_HEX < 0x03070000
```

Comment:
Is there a particular reason why this version number changed from `0x030700b0` to `0x03070000`?

Reply:
#5653 used `0x03070000`: https://github.com/obsproject/obs-studio/pull/5653/files#diff-07e241c573bec44c5491a58f0e65cf92d43af09a1306a864cb29240980f55759

But #6915 changed it for no given reason: https://github.com/obsproject/obs-studio/pull/6915/files#diff-07e241c573bec44c5491a58f0e65cf92d43af09a1306a864cb29240980f55759 maybe a typo 

And Python docs says (https://docs.python.org/3/c-api/init.html#c.PyEval_InitThreads):
> Changed in version 3.7: This function is now called by Py_Initialize(), so you don’t have to call it yourself anymore.

> Changed in version 3.7: The GIL is now initialized by Py_Initialize().

I will ask Pat off-thread.

Reply:
I look and this `b` just mean beta, and I see no reason to be so specific.

Reply:
In theory this needs to encompass all versions that contain the breaking change (and that includes the beta version). The real-life impact will probably small (only users of Python 3.7.0 Beta are affected), but semantically speaking _including_ the beta version is more correct.

Reply:
`0x03070000` is less than `0x030700b0`, so the beta is included.

Note: `f` is final release.

Reply:
The only requirement here is that the comparison needs to hit any version up to the one that still required manual initialisation of threads. If this is true for all 3.7 versions (including alphas) then `0x03070000` is fine.

Reply:
Alpha use `a` so less than beta.